TombLib/TombLib/Wad/TrLevels/TrLevel.cs

```diff
@@ -104,7 +104,7 @@ public void LoadLevel(string fileName, bool allowTRNGDecryption)
                 }
 
                 // Read the palette only for TR2 and TR3, TR1 has the palette near the end of the file
-                if (Version == TRVersion.Game.TR2 || Version == TRVersion.Game.TR3)
+                if (Version.Native() is TRVersion.Game.TR2 or TRVersion.Game.TR3)
```

Comment:
I am not sure why `LoadLevel` needs `Native()`. This function loads level files in original formats, and to my knowledge, no custom engines such as TRX/TRNG use custom version format, they use other means of versioning. Therefore, level files will always contain native version, unless TRX or TRNG developers will decide to change this convention.

Look in the beginning of `LoadLevel` function and see how `Version` field is derived.